tests/016-test-auto-setup-cleanup.sh

```diff
@@ -0,0 +1,34 @@
+source $SRCDIR/libtest.sh
+
```

Comment:
We will need a equivalent test for non docker/compatibility mode as well. Because this option is not specific to docker only. It should work with command line too.

Reply:
Feels a bit weird to be setting up storage "directly"¹ but choose `STORAGE_DRIVER=auto` right?  It feels like it's fine if those admins have `STORAGE_DRIVER=overlay2` in their config files...in the (IMO fairly unlikely now) event that we add a new driver, that'd be a pretty major event; not something to just have happen automatically right?

¹ Calling this "non-compatibility mode" feels weird, how about "direct mode"?

Reply:
Sure, we can call it `direct` mode.  If STORAGE_DRIVER=auto makes sense for docker, then it makes sense for direct mode also why not. Admin is asking tools to choose a driver based on what kernel is supporting. I mean how two cases are different.

Reply:
Let me take that back. I think in current form STORAGE_DRIVER=auto might not make much sense for direct mode. And reason being that it primarily is doing ftype check. Which should be done on the filesystem which will be used as upper layer by container run time. And in direct mode, tool has no idea which filesystem/dir container run time is planning to use for upper dirs.

For compatibility mode it *sort* of works because we have hardcoded location of docker dir /var/lib/docker. That will be broken too if docker is started with option `-g`.

So lets atleast block being able to specify STORAGE_DRIVER=auto in direct mode for now.